subsys/lorawan/CMakeLists.txt

```diff
@@ -0,0 +1,35 @@
+# SPDX-License-Identifier: Apache-2.0
+
+# lorawan.c depends on the include directories exposed by the loramac-node
+# library. Hence, if it exists then the source files are added to that otherwise
+# a library with same name is created.
+if(TARGET loramac-node)
+	set(ZEPHYR_CURRENT_LIBRARY loramac-node)
+else()
+	zephyr_library_named(loramac-node)
+endif()
+
+if(CONFIG_LORAMAC_REGION_AS923)
+	zephyr_compile_definitions(-DREGION_AS923)
```

Comment:
1. I think the `-D` should be omitted, so `zephyr_compile_definitions(REGION_AS923)`

2. You could also use this here:
```
zephyr_compile_definitions_ifdef(CONFIG_LORAMAC_REGION_AS923 REGION_AS923)
```

Reply:
Done!

Reply:
The first point is done, the second is not. Do you prefer

```
if(CONFIG_LORAMAC_REGION_AS923)
￼	zephyr_compile_definitions(REGION_AS923)
￼elseif(CONFIG_LORAMAC_REGION_AU915)
￼	zephyr_compile_definitions(REGION_AU915)
...
```

over

```
zephyr_compile_definitions_ifdef(CONFIG_LORAMAC_REGION_AS923
                                 REGION_AS923)
zephyr_compile_definitions_ifdef(CONFIG_LORAMAC_REGION_AU915
                                 REGION_AU915)
```

?

Reply:
Ack. Now it's done really :)